zk/lock.go

```diff
@@ -72,9 +83,9 @@ func (l *Lock) Lock() error {
 					return err
 				}
 			}
-		} else if err == nil {
+		case nil:
 			break
```

Comment:
this won't work the way it was intendet. 
break will only break switch statement.

switch should be removed or you need to add loop point

```
loop:
for i := 0; i < 3; i++ {
   switch:
   case nil:
      break loop
}
```
Example:
https://goplay.x1unix.com/snippet/xL1H5y8AvXB